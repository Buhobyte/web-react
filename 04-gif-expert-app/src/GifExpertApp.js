import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One-Punch', 'Dragon-Ball', 'Doraemon']);

    // const handleAdd = (e) => {
    //     // setCategories([...categories, 'Mazinger-Z']);
    //     setCategories(cats => [...cats, 'Mazinger-Z']); // puede recibir un callback 
    // }

    return (
        <>
            <h2>Git Expert App</h2>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <AddCategory setCategories={setCategories} />

            <ol>
                {
                    categories.map((category) => {
                        return <li key={category}>{category}  </li>;
                    })
                }
            </ol>
        </>
    )
}
