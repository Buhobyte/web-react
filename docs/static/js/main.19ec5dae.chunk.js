(this["webpackJsonpgif-expert-app"]=this["webpackJsonpgif-expert-app"]||[]).push([[0],{15:function(t,e,n){},18:function(t,e,n){"use strict";n.r(e);var c=n(1),r=n(7),a=n.n(r),i=(n(15),function(t){t&&t instanceof Function&&n.e(3).then(n.bind(null,19)).then((function(e){var n=e.getCLS,c=e.getFID,r=e.getFCP,a=e.getLCP,i=e.getTTFB;n(t),c(t),r(t),a(t),i(t)}))}),u=n(2),s=n(9),o=n(0),j=function(t){var e=t.setCategories,n=Object(c.useState)("Ingrese texto"),r=Object(u.a)(n,2),a=r[0],i=r[1];return Object(o.jsx)("form",{onSubmit:function(t){t.preventDefault(),a.trim().length>1&&(e((function(t){return[a].concat(Object(s.a)(t))})),i(""))},children:Object(o.jsx)("input",{type:"text",value:a,onChange:function(t){i(t.target.value)}})})},l=n(10),d=function(t){var e=t.title,n=t.url;return Object(o.jsxs)("div",{className:"item-card animate__fadeIn",children:[Object(o.jsx)("img",{alt:e,src:n}),Object(o.jsx)("p",{children:e})]})},f=n(6),p=n.n(f),g=n(8),b=function(){var t=Object(g.a)(p.a.mark((function t(e){var n,c,r,a,i,u;return p.a.wrap((function(t){for(;;)switch(t.prev=t.next){case 0:return n=e.category,"IeSlcE9JlBLhzpNlK7PPFQHfkvcg0mjY",c="https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(n),"&limit=10&api_key=").concat("IeSlcE9JlBLhzpNlK7PPFQHfkvcg0mjY"),t.next=5,fetch(c);case 5:return r=t.sent,t.next=8,r.json();case 8:return a=t.sent,i=a.data,u=i.map((function(t){var e;return{id:t.id,title:t.title,url:null===(e=t.images)||void 0===e?void 0:e.downsized_medium.url}})),t.abrupt("return",u);case 12:case"end":return t.stop()}}),t)})));return function(e){return t.apply(this,arguments)}}(),h=function(t){var e=t.category,n=function(t){var e=t.category,n=Object(c.useState)({data:[],loading:!0}),r=Object(u.a)(n,2),a=r[0],i=r[1];return Object(c.useEffect)((function(){b({category:e}).then((function(t){i({data:t,loading:!1})}))}),[e]),a}({category:e}),r=n.data,a=n.loading;return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h2",{children:e}),a&&Object(o.jsx)("p",{children:"Cargando..."}),Object(o.jsx)("div",{className:"card-grid animate__fadeIn",children:r.map((function(t){return Object(o.jsx)(d,Object(l.a)({},t),t.id)}))})]})},O=function(){var t=Object(c.useState)(["One Punch"]),e=Object(u.a)(t,2),n=e[0],r=e[1];return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h2",{children:"Gif Expert App"}),Object(o.jsx)("hr",{}),Object(o.jsx)(j,{setCategories:r}),Object(o.jsx)("ol",{children:n.map((function(t){return Object(o.jsx)(h,{category:t},t)}))})]})};a.a.render(Object(o.jsx)(O,{}),document.getElementById("root")),i()}},[[18,1,2]]]);
//# sourceMappingURL=main.19ec5dae.chunk.js.map