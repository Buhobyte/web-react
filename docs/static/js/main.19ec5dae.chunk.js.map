{"version":3,"sources":["reportWebVitals.js","components/AddCategory.js","components/GifGridItem.js","helpers/getGifs.js","components/GifGrid.js","hooks/useFetchGifs.js","GifExpertApp.js","index.js"],"names":["reportWebVitals","onPerfEntry","Function","then","getCLS","getFID","getFCP","getLCP","getTTFB","AddCategory","setCategories","useState","inputValue","setInputValue","onSubmit","e","preventDefault","trim","length","categories","type","value","onChange","target","GifGridItem","title","url","className","alt","src","dataGifAPI","a","category","encodeURI","fetch","resp","json","data","imgs","map","item","id","images","downsized_medium","GifGrid","loading","state","setState","useEffect","useFectchGifs","img","GifExpertApp","ReactDOM","render","document","getElementById"],"mappings":"8KAYeA,G,MAZS,SAAAC,GAClBA,GAAeA,aAAuBC,UACxC,6BAAqBC,MAAK,YAAkD,IAA/CC,EAA8C,EAA9CA,OAAQC,EAAsC,EAAtCA,OAAQC,EAA8B,EAA9BA,OAAQC,EAAsB,EAAtBA,OAAQC,EAAc,EAAdA,QAC3DJ,EAAOH,GACPI,EAAOJ,GACPK,EAAOL,GACPM,EAAON,GACPO,EAAQP,Q,qBCJDQ,EAAc,SAAC,GAAuB,IAArBC,EAAoB,EAApBA,cAE1B,EAAoCC,mBAAS,iBAA7C,mBAAOC,EAAP,KAAmBC,EAAnB,KAeA,OAEI,aADA,CACA,QAAMC,SAXW,SAACC,GAClBA,EAAEC,iBAEEJ,EAAWK,OAAOC,OAAS,IAC3BR,GAAc,SAAAS,GAAU,OAAKP,GAAL,mBAAmBO,OAC3CN,EAAc,MAMlB,SAEI,uBAAOO,KAAK,OAAOC,MAAOT,EAAYU,SAjBpB,SAACP,GACvBF,EAAcE,EAAEQ,OAAOF,a,QCNlBG,EAAc,SAAC,GAAoB,IAAlBC,EAAiB,EAAjBA,MAAOC,EAAU,EAAVA,IACjC,OACI,sBAAKC,UAAU,4BAAf,UACI,qBAAKC,IAAKH,EAAOI,IAAKH,IACtB,4BAAID,Q,uBCNHK,EAAU,uCAAG,uCAAAC,EAAA,6DAAQC,EAAR,EAAQA,SACd,mCACVN,EAFgB,iDAEgCO,UAAUD,GAF1C,6BACN,oCADM,SAGHE,MAAMR,GAHH,cAGhBS,EAHgB,gBAICA,EAAKC,OAJN,uBAIdC,EAJc,EAIdA,KAEFC,EAAOD,EAAKE,KAAI,SAACC,GAAU,IAAD,EAC5B,MAAO,CACHC,GAAID,EAAKC,GACThB,MAAOe,EAAKf,MACZC,IAAG,UAAEc,EAAKE,cAAP,aAAE,EAAaC,iBAAiBjB,QAVrB,kBAcfY,GAde,4CAAH,sDC6BRM,EAxBC,SAAC,GAAkB,IAAhBZ,EAAe,EAAfA,SAEf,ECJyB,SAAC,GAAkB,IAAhBA,EAAe,EAAfA,SAE5B,EAA0BrB,mBAAS,CAAC0B,KAAM,GAAIQ,SAAS,IAAvD,mBAAOC,EAAP,KAAcC,EAAd,KAYA,OATAC,qBAAU,WAENlB,EAAW,CAAEE,aAAY7B,MAAM,SAAAmC,GAC3BS,EAAS,CAAEV,KAAMC,EAAMO,SAAS,SAGrC,CAACb,IAGGc,EDV2BG,CAAc,CAAEjB,aAApCU,EAAd,EAAQL,KAAcQ,EAAtB,EAAsBA,QAEtB,OAEI,qCACI,6BAAKb,IACJa,GAAW,4CAEZ,qBAAKlB,UAAU,4BAAf,SAEQe,EAAOH,KAAI,SAACW,GAAD,OAAS,cAAC,EAAD,eAA8BA,GAAZA,EAAIT,aEbjDU,EAAe,WAExB,MAAoCxC,mBAAS,CAAC,cAA9C,mBAAOQ,EAAP,KAAmBT,EAAnB,KAOA,OACI,qCACI,gDACA,uBAEA,cAAC,EAAD,CAAaA,cAAeA,IAC5B,6BAEQS,EAAWoB,KAAI,SAACP,GAAD,OACX,cAAC,EAAD,CAAwBA,SAAUA,GAApBA,YChBtCoB,IAASC,OAEL,cAAC,EAAD,IAEFC,SAASC,eAAe,SAM1BvD,M","file":"static/js/main.19ec5dae.chunk.js","sourcesContent":["const reportWebVitals = onPerfEntry => {\n  if (onPerfEntry && onPerfEntry instanceof Function) {\n    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {\n      getCLS(onPerfEntry);\n      getFID(onPerfEntry);\n      getFCP(onPerfEntry);\n      getLCP(onPerfEntry);\n      getTTFB(onPerfEntry);\n    });\n  }\n};\n\nexport default reportWebVitals;\n","import React, { useState } from 'react'\r\nimport PropTypes from 'prop-types'\r\n\r\nexport const AddCategory = ({ setCategories }) => {\r\n\r\n    const [inputValue, setInputValue] = useState('Ingrese texto');\r\n\r\n    const handleInputChange = (e) => {\r\n        setInputValue(e.target.value);\r\n    }\r\n\r\n    const handleSubmit = (e) => {\r\n        e.preventDefault();\r\n\r\n        if (inputValue.trim().length > 1) {\r\n            setCategories(categories => [inputValue,...categories])\r\n            setInputValue('');\r\n        }\r\n    }\r\n\r\n    return (\r\n        // <> no es necesario el fragment ya que form sirve de fragmentador o agrupador\r\n        <form onSubmit={handleSubmit}>\r\n            {/* <h2>{inputValue}</h2> */}\r\n            <input type='text' value={inputValue} onChange={handleInputChange}></input>\r\n        </form>\r\n        // </>\r\n    )\r\n\r\n\r\n}\r\n\r\nAddCategory.propTypes = {\r\n    setCategories: PropTypes.func.isRequired\r\n}\r\n","import React from 'react'\r\n\r\nexport const GifGridItem = ({ title, url }) => {\r\n    return (\r\n        <div className='item-card animate__fadeIn'>\r\n            <img alt={title} src={url}></img>\r\n            <p>{title}</p>\r\n        </div>\r\n    )\r\n}\r\n","export const dataGifAPI = async ({category}) => {\r\n    const api_key = 'IeSlcE9JlBLhzpNlK7PPFQHfkvcg0mjY';\r\n    const url = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(category)}&limit=10&api_key=${api_key}`;\r\n    const resp = await fetch(url);\r\n    const { data } = await resp.json();\r\n\r\n    const imgs = data.map((item) => {\r\n        return {\r\n            id: item.id,\r\n            title: item.title,\r\n            url: item.images?.downsized_medium.url,\r\n\r\n        }\r\n    })\r\n    return imgs;\r\n}\r\n","import React from 'react'\r\nimport PropTypes from 'prop-types'\r\nimport { GifGridItem } from './GifGridItem';\r\nimport { useFectchGifs } from '../hooks/useFetchGifs';\r\n\r\nconst GifGrid = ({ category }) => {\r\n\r\n    const { data: images, loading } = useFectchGifs({ category });\r\n\r\n    return (\r\n\r\n        <>\r\n            <h2>{category}</h2>\r\n            {loading && <p>Cargando...</p>}\r\n\r\n            <div className=\"card-grid animate__fadeIn\">\r\n                {\r\n                    images.map((img) => <GifGridItem key={img.id} {...img} />) // pasamos todos las propiedades incluido img\r\n                }\r\n            </div>\r\n\r\n        </>\r\n    )\r\n}\r\n\r\nGifGrid.propTypes = {\r\n    categories: PropTypes.array\r\n}\r\n\r\nexport default GifGrid\r\n","import { useEffect, useState } from \"react\";\r\nimport { dataGifAPI } from \"../helpers/getGifs\";\r\n\r\nexport const useFectchGifs = ({ category }) => {\r\n\r\n    const [state, setState] = useState({data: [], loading: true});\r\n\r\n    // se ejecuta solo una vez y no renderiza nuevamente todo el componente\r\n    useEffect(() => {\r\n\r\n        dataGifAPI({ category }).then((imgs => {\r\n            setState({ data: imgs, loading: false });\r\n        }));\r\n\r\n    }, [category]);\r\n    // add category becouse if category changes then useEffect needs to change\r\n\r\n    return state;\r\n}","import React, { useState } from 'react';\r\nimport { AddCategory } from './components/AddCategory';\r\nimport GifGrid from './components/GifGrid';\r\n\r\nexport const GifExpertApp = () => {\r\n\r\n    const [categories, setCategories] = useState(['One Punch']);\r\n\r\n    // const handleAdd = (e) => {\r\n    //     // setCategories([...categories, 'Mazinger-Z']);\r\n    //     setCategories(cats => [...cats, 'Mazinger-Z']); // puede recibir un callback \r\n    // }\r\n\r\n    return (\r\n        <>\r\n            <h2>Gif Expert App</h2>\r\n            <hr />\r\n            {/* <button onClick={handleAdd}>Agregar</button> */}\r\n            <AddCategory setCategories={setCategories} />\r\n            <ol>\r\n                {\r\n                    categories.map((category) =>\r\n                        <GifGrid key={category} category={category} />\r\n                    )\r\n                }\r\n            </ol>\r\n\r\n\r\n        </>\r\n    )\r\n}\r\n","import React from 'react';\nimport ReactDOM from 'react-dom';\nimport './index.css';\nimport reportWebVitals from './reportWebVitals';\nimport { GifExpertApp } from './GifExpertApp';\n\nReactDOM.render(\n  // <React.StrictMode>\n    <GifExpertApp />,\n  // </React.StrictMode>,\n  document.getElementById('root')\n);\n\n// If you want to start measuring performance in your app, pass a function\n// to log results (for example: reportWebVitals(console.log))\n// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals\nreportWebVitals();\n"],"sourceRoot":""}