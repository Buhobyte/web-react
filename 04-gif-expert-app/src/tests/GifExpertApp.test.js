import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp.js'
describe('Probar componente ExpertAPP', () => {

    test('deberÃ­a mostrarse el componente GitExpertApp', () => {
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Dragon Ball']

        const wrapper = shallow(<GifExpertApp defaultCatagories={categories} />)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('GifGrid').length).toBe(2);

    })


})

