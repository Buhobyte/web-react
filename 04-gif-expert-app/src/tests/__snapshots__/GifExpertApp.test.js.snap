// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar componente ExpertAPP debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Probar componente ExpertAPP deber√≠a mostrarse el componente GitExpertApp 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
